### ISSUE TITLE
angular/ember/etc examples?

### ISSUE THREAD

**Issue by jayphelps** [redux-observable/redux-observable#7]
Nothing about this is React specific at all, and with momentum growing with angular-redux and ember-redux it would be very beneficial for usage examples with them added to the docs as well as in the [examples/](examples/) directory.

https://github.com/angular-redux/ng-redux (angular 1)
https://github.com/angular-redux/ng2-redux (angular 2)
https://github.com/toranb/ember-redux

**Comment by blesh** [comment]
cc/ @robwormald

**Comment by HighOnDrive** [comment]
This has me thinking of the case where Rethink Horizon is the store and potential usage in an app that does not require Redux. It would be great if only the React example (not the core API) had the Redux dependency?

**Comment by jayphelps** [comment]
@HighOnDrive I'm not familiar with Rethink Horizon and can't find anything definitive from a google search. Can you clarify the library and use case?

To your point, you're right that we don't actually _need_ to depend on redux; we in fact never even import it in the library code, only in the tests + examples. Since it's just a peer dependency, in npm@3 they are optional, it doesn't fail to install--just gives the the annoying warning. I realize perhaps this warning isn't ideal, if you can think of some genuine use cases outside of redux, let's brainstorm some alternatives.

**Comment by HighOnDrive** [comment]
Horizon by RethinkDB is a new collaborative service for the client that uses RxJS under the hood. Here is more info on this new Firebase alternative: https://github.com/rethinkdb/horizon

The ability to dispatch and cancel actions in such a flexible way looks ideal for a general state driven use case. I'm thinking of a feature driven app where the features are activated and deactivated, this works somewhat like a state machine. Each feature is it's own piece of state that defines the actions to be used. When a feature is activated it's actions get dispatched and when another feature is activated the prior feature has it's actions canceled.

When a specific feature state is activated the dispatch function would be injected with the names/ids of the right actions to run. Taking this further a feature state can also define what actions multiple dispatchers should run. I see no reason why the dispatch mechanizm could not be used for multiple processors, for instance processors handling data modification or rendering data to a view. Actions to me are just functions so they could even be view partials.

I see this new API as an ideal scalable action dispatch solution for larger apps, where actions sets can be defined as feature states incrementally, which then in turn transform the reactive loop with many alternate behaviors/actions.

**Comment by e-schultz** [comment]
@jayphelps @blesh I'm working on ng2-redux, and could help with an Angular 2 example,

**Comment by jayphelps** [comment]
@e-schultz please do! 👍

**Comment by HighOnDrive** [comment]
Still hoping for a universal solution, something that can work alongside any framework, React, Vue or whatever. Noticed that Horizon was released today: http://rethinkdb.com/blog/horizon-release

**Comment by jayphelps** [comment]
@HighOnDrive we're happy to consider such a proposition, but we need some concrete example of how it would work.

Could you do me a favor and create a new issue to track that request and include some code on how you think it would work? Not the implementation, the consuming side of things...mainly how a person would programmatically choose to use redux for dispatching or some other alternative. Thanks!

**Comment by e-schultz** [comment]
Traveling right now, but will try and get an example done soon.

**Comment by HighOnDrive** [comment]
@jayphelps Just took time to look redux-observable over again, it looks like a step toward the kind of pattern I'm ultimately looking for. TBH, I'm fascinated by dispatch/injection systems and seeing higher order patterns emerge from RxJS (besides that restrictive one from CycleJS) is a good thing. I'll keep working my own dispatch solution and for now just see redux-observable as a nice pattern (separating it from React/Redux in my mind), for my own learning :)

I still advocate that RxJS could stand a few more patterns on it's own and it's a drag to see them being dedicated to specific frameworks right off the hop, because I really see RxJS as being quite beyond the frameworks. In any case major RxJS patterns should be useful in any framework.

**Comment by sebald** [comment]
@jayphelps Here is a vanilla version for ng1 (w/o ng-redux): http://jsbin.com/buxayow/1/edit?js,output

ng-redux is missing because I only saw the missing ng1 Example in the readme and discovered the issue only after I already wrote the jsbin :confused: I personally do not see the need for the library (we don't use it in our software, only angular+redux+redux-observable 😃 ).
If you rather have a version with ng-redux, let me know.

PS: I switched the compiler to TypeScript because jsbin wouldn't let me write string literals :(

**Comment by jayphelps** [comment]
@sebald thank you very much!!

**Comment by sebald** [comment]
Your welcome!

---

I couldn't let it go and had to try `ng-redux` 😳 

There is a very high propability that I'm just to dumb, but I couldn't get it to work with `redux-observable`. Here is the jsbin with some debug lines: http://jsbin.com/cakobe/edit?js,console,output

`ng-redux` is (web)packed including `redux` and it only seems to work with the old `createStore` syntax (Array instead of `applyMiddleware`). The `pingReducer` is not correctly called. Maybe @wbuchwalter could take a loot at the bin.

**Comment by br1anchen** [comment]
hi @sebald , I try to fix the error in your jsbin, it should work now: [https://gist.github.com/anonymous/67bdb57152096901f1849d6da7cce5c8](url).

It seems like just wrong scripts imported in index.html from script tag for `redux-observable`, I checked with ng1 example to make it work.

I have question regarding to use angular DI in `Epic` function as side effect (mainly for `$http` request), is there any way to use `Epic` function rather than pure function? Since `Epic` need to be applied/created by `createEpicMiddleware`, any idea how to use other ng service/factory inside the `Epic` functions?

**Comment by blesh** [comment]
I have a primitive [Angular 2 example here](http://plnkr.co/edit/vfrj3duOloEF8XZjMS5m?p=preview)

**Comment by e-schultz** [comment]
Used redux-observable in a demo app I built for a conference: https://github.com/e-schultz/ng-summit-2016-shelter

**Comment by ethanrubio** [comment]
I have an ember twiddle version of this working here: https://ember-twiddle.com/347867726ecf8e6843b5def6a201ec6e?openFiles=components.ping-pong.js%2C

Would this be acceptable to add to the README? Or should it be a jsbin?
